projects/js-toolkit/packages/portal-base/src/build/themeSpritemap.ts

```diff
@@ -0,0 +1,172 @@
+/**
+ * SPDX-FileCopyrightText: © 2021 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: LGPL-3.0-or-later
+ */
+
+/* eslint-disable @liferay/no-dynamic-require */
+
+import {
+	ClientExtensionConfigJson,
+	FilePath,
+	Project,
+	ThemeSpritemapBuildOptions,
+	format,
+} from '@liferay/js-toolkit-core';
+import fs from 'fs';
+import globby from 'globby';
+import JSZip from 'jszip';
+import path from 'path';
+
+import abort from '../util/abort';
+
+const {error, info, print} = format;
+
+const HEADER_REGEXP = /<!--(.*)-->/s;
+
+const SPRITEMAP_FILE_NAME = 'spritemap.svg';
+
+export default async function customElement(project: Project): Promise<void> {
+	const {enableSVG4Everybody = false, extendClay = false} = project.build
```

Comment:
I would move defaults to the Project "framework". The idea is that any code using `project` can assume it doesn't need to do any defaults, normalization or even decision handling.

In the case of "decision" I'm referring to: if we can remove an `if` by returning a different value from the config, for example, then move that logic to the config Project framework.

Example: instead of adding an `if` that decides an output file name, add an output property in the config and decide that name in the Project side. That way the build code remains cleaner and simpler. Of course this cannot be done for everything, but I've found it's a good pattern in the past...

Reply:
Also, when normalizing the config I try to canonicalize null, undefined, and that stuff to the most restrictive type possible. So, if I can use a boolean, I  normalize it so that it is not null or undefined, when reading the config.

Reply:
I removed the defaults, as they weren't even needed, but I'm not really sure what you mean by the examples...

> we can remove an `if`

Any "if"? or was there a specific one in mind?